src/Components/Comments.jsx

```diff
@@ -1,10 +1,17 @@
+import { useEffect } from "react";
 import AddComment from "./AddComment";
 import CommentCard from "./CommentCard";
+import { getCommentsByArticle } from "../../api";
 
 
 const Comments = ({states}) => {
     const [article_id, comments, setComments, showComments, setShowComments] = states;
-   
+    
+    useEffect(()=> {
+        getCommentsByArticle(article_id).then(({comments}) => {
+            setComments(comments)
+        })
+    }, [comments])
```

Comment:
Line `12` and line `14` form an infinite loop of endlessly fetching comments 🔁 🔁 🔁 

_this is why it's updating the list to remove the comment after one gets deleted_